plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -5371,7 +5371,7 @@ protected AttachIsoAnswer execute(AttachIsoCommand cmd) {
                                         "Failed to unmount vmware-tools installer ISO as the corresponding CDROM device is locked by VM. Please unmount the CDROM device inside the VM and ret-try.");
                             }
                         } catch (Throwable e) {
-                            vmMo.detachIso(null);
+                            vmMo.detachIso(null, cmd.isForce());
```

Comment:
_cmd.isForced()_ here

Reply:
@sureshanaparti isForce() is a function of AttachIsoCommand(not AttachCommand), that already existed, and I'm just calling that.